### DISCUSSION TITLE
Looks like mapping UDP ports isn't supported. Any known workarounds?

### DISCUSSION THREAD

**Discussion by wolfman42** [testcontainers/testcontainers-dotnet#1216] [Q&A] [answered]
Implementation of `GetMappedPublicPort()` hard-codes the `tcp` suffix
See https://github.com/testcontainers/testcontainers-dotnet/blob/b1ad2459a163cc29423ea72df25950ee015c73aa/src/Testcontainers/Containers/DockerContainer.cs#L239

Would be nice if something like this was possible:
`container.GetMappedPublicPort("161/udp");`

**Comment by HofmeisterAn** [comment, marked as answer]
Unfortunately, it looks like this is not supported yet. Sorry about that. I don't know why I haven't added support for the other protocols as well. We need to overload the method.

**Reply by wolfman42** [reply]
Too bad. Are you planning to work on this or should I give it a go?
What API changes / additions do you suggest?

**Reply by wolfman42** [reply]
I went ahead and created issue #1218 along with PR #1219.
Would appreciate this getting merged soon.

**Reply by wolfman42** [reply]
Unit tests are passing now. Would appreciate your review.

**Reply by HofmeisterAn** [reply]
I will take a look at it sometime today.
